R/observations.R

```diff
@@ -62,12 +63,12 @@ observation_impute <- function(W_kernel, S, x_train, x_test, w_threshold = .7, n
   # Remove training data with small weight
   knms <- c("index_s", "weight")
   data.table::setkeyv(dt_melt, knms)
-  dt_melt[, weight := weight / sum(weight), index_s]
+  dt_melt[, weight := weight / sum(weight), by = "index_s"]
```

Comment:
Just out of curiosity: Why is this better? Is it along the lines that new version would throw an error if index_s is not in dt_melt, while the old one wouldn't if index_s was defined outside of dt_melt?

Reply:
This was basically to avoid the notes. But I also checked your question, and the answer to that is yes (!). So I'll need to go over tomorrow and see if there is more cases where we should add this as well.